modules/test/playwright/tests/style-book-web/previewStyleBookOnDisplayPageTemplates.spec.ts

```diff
@@ -0,0 +1,247 @@
+				.getByRole('heading', {name: 'Heading Example'});
+
+			const fragmentColor = await heading.evaluate((element) => {
+				const computedStyle = window.getComputedStyle(element);
+
+				return computedStyle.color;
+			});
+
+			expect(fragmentColor).toBe(colorRGB);
+		};
+
+		await page.getByRole('button', {name: 'Color System'}).click();
+
+		await page.getByRole('menuitem', {name: 'General'}).click();
+
+		await selectOptionAndUpdateColor(
+			displayPageTemplateNameBlogsEntry,
+			displayPageTemplateNameBlogsEntry,
+			'#995511',
+			'rgb(153, 85, 17)'
+		);
+
+		await selectOptionAndUpdateColor(
+			displayPageTemplateNameBlogsEntry,
+			displayPageTemplateNameWebContent,
+			'#666666',
+			'rgb(102, 102, 102)'
+		);
+	});
+});
+
+test('Use the view more button to select a display page template from the select modal and change preview page', async ({
+	displayPageTemplatesPage,
+	page,
+	site,
+	styleBooksPage,
+}) => {
+	const displayPageTemplateNameBlogsEntry = getRandomString();
+	const displayPageTemplateNameCategory = getRandomString();
+	const displayPageTemplateNameDocuments = getRandomString();
+	const displayPageTemplateNameWebContent1 = getRandomString();
+	const displayPageTemplateNameWebContent2 = getRandomString();
+	const styleBookName = getRandomString();
+
+	await test.step('Create and publish multiple display page templates', async () => {
+		await displayPageTemplatesPage.goto(site.friendlyUrlPath);
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Web Content',
+			contentType: 'Web Content Article',
+			name: displayPageTemplateNameWebContent1,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Web Content',
+			contentType: 'Web Content Article',
+			name: displayPageTemplateNameWebContent2,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentType: 'Blogs Entry',
+			name: displayPageTemplateNameBlogsEntry,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentSubtype: 'Basic Document',
+			contentType: 'Document',
+			name: displayPageTemplateNameDocuments,
+		});
+
+		await displayPageTemplatesPage.createTemplate({
+			contentType: 'Category',
+			name: displayPageTemplateNameCategory,
+		});
+	});
+
+	await test.step('Click on the view more button to select a new display page template from the modal', async () => {
+		await styleBooksPage.goto(site.friendlyUrlPath);
+
+		await styleBooksPage.create(styleBookName);
+
+		await clickAndExpectToBeVisible({
+			autoClick: true,
+			target: page.getByRole('button', {name: 'More'}),
+			trigger: page.getByRole('button', {
+				name: displayPageTemplateNameCategory,
+			}),
+		});
+
+		await page
+			.frameLocator('iframe[title="Select"]')
+			.getByRole('button', {
+				name: 'Select ' + displayPageTemplateNameWebContent1,
+			})
+			.click();
+
+		const loadingAnimation = page.locator('.loading-animation').first();
+
+		await loadingAnimation.waitFor();
```

Comment:
Blank line after 238

Reply:
In this case, I believe we can maintain it like this since it is alphabetically ordered, following these rules:

- If the order of the lines doesn't matter and they can be sorted alphabetically, then keep them together. (IE: no line break)
- If order matters and you can't alphabetically sort the lines then add a line break between each line.

Reply:
Actually, just now I noticed, why the code waits twice? Can't it just wait for the element to be hidden?

Reply:
It's duplicated because first, we have to wait for the loading to be displayed and then wait to be hidden.
If pass the only the hidden one, it will fail.